python/tests/README.md

```diff
@@ -0,0 +1,69 @@
+
+Given a JSON file as input, the Client sends a query to the Server and prints the predicted output. 
+
+This server is configured for JSON input and output. 
+
+### [`test_transform_process_arrow.py`](test_transform_process_arrow.py)
+
+Similar to `test_transform_process.py`, checks if a DataVec transformation can be configured, but with the server configured to return Arrow output instead of JSON. 
+
+### [`test_python_serving.py`](test_python_serving.py)
+
+Tests whether Python pipelines can be served. Takes as input a NumPy array saved in `../data/input-0.npy` and performs a simple operation on the model inputs in a [PythonPipelineStep](../konduit/inference.py#L1221-L1394).
+
+The input and output data types of this server are configured to be NUMPY. 
+
+### [`test_bert_serving.py`](test_bert_serving.py)
+
+Similar to `test_start.py`, creates a configuration with a TensorFlow BERT model loaded with [TensorFlowConfig](../konduit/inference.py#L656-L720), but also prints the predicted output given NumPy arrays. 
+
+The input and output data types of this server are configured to be NUMPY. 
+
+## Saving configurations 
+
+### [`test_json.py`](test_json.py)
+
+After configuring a server, saves server configuration as JSON, loads it again, and checks if the existing configuration steps have the same length as the configuration loaded from the JSON file. 
+
+Two server configurations are defined: 
+- Similar to `test_start.py`, `test_json_compare()` configures a server with a TensorFlow model loaded with [TensorFlowConfig](../konduit/inference.py#L656-L720), 
+- `test_python_serde()` configures a server with a [PythonPipelineStep](../konduit/inference.py#L1221-L1394) similar to `test_python_serving.py`.
+
+
+## Client conversion and encoding 
+
+### [`test_client_serde.py`](test_client_serde.py)
+
+#### `test_multipart_encode()`
+Tests the following methods for the [Client](../konduit/client.py#L10-L139) class: 
+- [`_convert_numpy_to_binary()`](../konduit/client.py#L72-L76): converts inputs into a dictionary (key-value pairs) with binary data as values.
+- [`_convert_multi_part_inputs()`](../konduit/client.py#L100-L108): encodes the dictionary from `convert_numpy_to_binary()` in a multipart request body.
+- [`_encode_multi_part_input()`](../konduit/client.py#L85-L97): decodes the multipart response into binary data or, depending on the output type, converts into the corresponding output class.
+- [`_convert_multi_part_output()`](../konduit/client.py#L110-L129): converts output returned by the server into the output type requested by the client.
```

Comment:
Maybe revise these lines.
1.  _convert_numpy_to_binary() just converts a numpy array into binary data. It doesn't have anything to do with key-value pairs
2. _convert_multi_part_inputs() takes a dictionary of key value pairs and reformat the values into binary if they're not so that they can be encoded to multipart request body later
3. __encode_multi_part_input() does the actual encoding from the output taken from _convert_multi_part_inputs
4. _convert_multi_part_output() decodes the output returned by the server into the output type requested by the client.